llvm/utils/filecheck_lint/filecheck_lint.py

```diff
@@ -228,7 +237,8 @@ def find_best_match(typo):
         )
 
     potential_directives = find_potential_directives(content)
-
+    # cache score and best_match to skip recalculating
+    checked_potential_directives = dict()
```

Comment:
nit: it's not super obvious to me what this variable contains based on its name.

How about `score_and_best_match_for_potential_directive`? (A bit of a mouthful.)